### ISSUE TITLE
False Positive in Redundant Conditional?

### ISSUE THREAD

**Issue by senfi** [oclint/oclint#182]
Take this code: 

```
-(BOOL)doSomethingWithParameter:(NSString*) param1
                       andAnotherParameter:(NSTimeInterval) param2
{
    Something* mySomething = nil;

    //check if parameter are cool and we are ready to go
    if ([self checkParameter:param1 andParameter:param2)
    {
        mySomething = [[Something alloc] initWithParameter:param1
                            andAnotherParameter: param2];
        //do something with mySomething
    }

    //if mySomething != nil, return true bool
    return (mySomething ? YES : NO);
    //the warning in the line of the return: Redundant conditional operator P3
}
```

I can't see, why this is redundant. This function is returning a BOOL, so  `return mySomething` would look strange too. Please correct me if i'm wrong, but i think this is a false positive.
